src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/CoverageModelCopyRatioEmissionProbabilityCalculator.java

```diff
@@ -0,0 +1,257 @@
+     *
+     * TODO github/gatk-protected issue #854
+     *
+     * @implNote This is a naive implementation where the variance of log bias ($\Psi$) is
+     * ignored altogether. This routine must be improved by performing a few-point numerical
+     * integration of:
+     *
+     *      \int_{-\infty}^{+\infty} db Poisson(n | \lambda = d*c*exp(b) + eps*d)
+     *                                                           * Normal(b | \mu, \Psi)
+     *
+     * @param emissionData an instance of {@link CoverageModelCopyRatioEmissionData}
+     * @param copyRatio copy ratio on which the emission probability is conditioned on
+     * @return a double value
+     */
+    private double logLikelihoodPoisson(@Nonnull CoverageModelCopyRatioEmissionData emissionData,
+                                        double copyRatio) {
+        final double multBias = FastMath.exp(emissionData.getMu());
+        final double readDepth = emissionData.getCopyRatioCallingMetadata().getSampleCoverageDepth();
+        final double poissonMean = readDepth * copyRatio * multBias + emissionData.getPoissonAdditiveNoise();
+        return new PoissonDistribution(null, poissonMean, PoissonDistribution.DEFAULT_EPSILON,
+                PoissonDistribution.DEFAULT_MAX_ITERATIONS).logProbability(emissionData.getReadCount());
+    }
+
+    private static double getUnnormalizedLogProbabilityLaplaceApproximation(final double readCount,
+                                                                            final double mu,
+                                                                            final double psi) {
+        final double variance = psi + 1.0 / readCount;
+        final double logReadCount = FastMath.log(readCount);
+        return - logReadCount - 0.5 * (LOG_2PI + FastMath.log(variance) +
+                FastMath.pow(logReadCount - mu, 2) / variance);
+    }
+
+    @VisibleForTesting
+    public static double getLogProbabilityLaplaceApproximationMass(final double mu, final double psi) {
+        if (mu > MAXIMUM_ALLOWED_MU) { /* the log norm factor is guaranteed to be smaller than 1e-4 */
+            return 0.0;
+        } else {
+            final double muTrunc = FastMath.max(MINIMUM_ALLOWED_MU, mu);
+            final double psiTrunc = FastMath.max(MINIMUM_ALLOWED_PSI, FastMath.min(MAXIMUM_ALLOWED_PSI, psi));
+            return logNormFactorSpline.value(muTrunc, psiTrunc);
+        }
+    }
+
+    private static double[] loadDoubleArrayTable(final InputStream inputStream) {
+        final Scanner reader = new Scanner(inputStream);
+        final List<Double> data = new ArrayList<>();
+        while (reader.hasNextLine()) {
+            data.add(Double.parseDouble(reader.nextLine()));
+        }
+        return data.stream().mapToDouble(d -> d).toArray();
+    }
+
+    private static double[][] loadDouble2DArrayTable(final InputStream inputStream) {
+        final Scanner reader = new Scanner(inputStream);
```

Comment:
David suggested using `org.broadinstitute.hellbender.utils.Utils.stream` from gatk. I have put a todo for now.